tensorflow/contrib/learn/python/learn/io/data_feeder.py

```diff
@@ -505,5 +507,89 @@ def _feed_dict_fn():
       encoded_out[np.arange(out.size), out] = 1
       return {input_placeholder.name: inp,
               output_placeholder.name: encoded_out}
+    return _feed_dict_fn
+
+
+class HDF5DataFeeder(object):
+  """Data feeder for TF trainer that reads data from HDF5 format.
+
+  HDF5 is a popular format to store complicated datasets (alternative to proto files).
+  This data feeder performs random samples on hdf5 data.
+
+  Parameters:
+    X: iterator that returns for each element, returns features.
+    y: iterator that returns for each element, returns 1 or many classes /
+      regression values.
+    n_classes: indicator of how many classes the target has.
+    batch_size: Mini batch size to accumulate.
+    random_state: random state for RNG. Note that it will mutate so use a
+      int value for this if you want consistent sized batches.
+
+  Attributes:
+    X: input features.
+    y: input target.
+    n_classes: number of classes.
+    batch_size: mini batch size to accumulate.
+    input_shape: shape of the input.
+    output_shape: shape of the output.
+    input_dtype: dtype of input.
+    output_dtype: dtype of output.
+  """
+  def __init__(self, X, y, n_classes, batch_size, random_state=None):
+    # TODO(terrytangyuan): make this compatible for unsupervised approaches
+    self.X = X
+    self.y = y
+    # TODO(terrytangyuan): save column names
+    self.n_classes = n_classes
+
+    X_shape = X.shape
+    y_shape = y.shape
+    self.input_shape, self.output_shape = _get_in_out_shape(X_shape, y_shape,
+                                                            n_classes,
+                                                            batch_size)
+    self.input_dtype, self.output_dtype = np.float32, np.float32
+    if random_state is None:
+      self.random_state = 66
+    else:
+      self.random_state = random_state
+    self.batch_size = batch_size
+
+  def get_feed_params(self):
+    """Function returns a dict with data feed params while training.
+
+    Returns:
+      A dict with data feed params while training.
+    """
+    return {'batch_size': self.batch_size}
+
+  def get_feed_dict_fn(self, input_placeholder, output_placeholder):
+    """Returns a function, that will sample data and provide it to placeholders.
+
+    Args:
+      input_placeholder: tf.Placeholder for input features mini batch.
+      output_placeholder: tf.Placeholder for output targets.
 
+    Returns:
+      A function that when called samples a random subset of batch size
+      from X and y.
+    """
+    def _feed_dict_fn():
+      sample_indices = np.random.randint(0, self.input_shape[0], self.batch_size)
```

Comment:
this code seems similar to regular data feed, I wonder if we can pull out a function or two.